docs/usage/topology_aware_routing.md

```diff
@@ -0,0 +1,93 @@
+
+The `TopologyAwareRouting` feature relies on the Kubernetes feature [`TopologyAwareHints`](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/).
+
+##### EndpointSlice Hints mutating webhook
+
+The component that is responsible for providing hints in the EndpointSlices resources is the [kube-controller-manager](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/), in particular this is the [EndpointSlice controller](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/). However, there are several drawbacks with the TopologyAwareHints feature that don't allow us to use it in its native way:
+- The algorithm in the EndpointSlice controller is based on a CPU-balance heuristic. From the TopologyAwareHints documentation:
+   > The controller allocates a proportional amount of endpoints to each zone. This proportion is based on the allocatable CPU cores for nodes running in that zone. For example, if one zone had 2 CPU cores and another zone only had 1 CPU core, the controller would allocate twice as many endpoints to the zone with 2 CPU cores.
+
+   In case it is not possible to achieve a balanced distribution of the endpoints, as a safeguard mechanism the controller removes hints from the EndpointSlice resource.
+   In our setup the clients and the servers are well-known and usually the traffic a component receives does not depend on the zone's allocatable CPU. 
+   Most of the components deployed by Gardener are scaled automatically by VPA (in most of the cases). In case of an overload of a replica, the VPA should provide and apply enhanced CPU and memory resources. Additionally, Gardener uses the cluster-autoscaler to upscale/downscale Nodes dynamically. Hence, it is not possible to ensure a balanced of allocatable CPU accross the zones.
+- TopologyAwareHints does not work at low-endpoint counts.
+- Hints provided by the EndpointSlice controller are not deterministic. With cluster-autoscaler running and load increasing in the next moment hints can be removed. There is no option to enforce the zone-level topology.
+
+For more details, see the following issue [kubernetes/kubernetes#113731](https://github.com/kubernetes/kubernetes/issues/113731).
+
+To circumvent these issues with the EndpointSlice controller, a mutating webhook in the resource-manager assigns hints to EndpointSlice resources. For each enpoint in the EndpointSlice it sets the endpoint's hints to the endpoint's zone. The webhook overwrites the hints provided by the EndpointSlice controller in kube-controller-manager. For more details, see the [webhook's documentation](../concepts/resource-manager.md#endpointslice-hints).
+
+##### kube-proxy
+
+By default, with kube-proxy running in `iptables` mode traffic is distributed randomly across all endpoints, regardless of where it originates from. In a 3 zone cluster, traffic is more likely to go to another zone than to stay in the current zone.
```

Comment:
Q: Does this also work with `IPVS`? If not then we'd have a dependency to how Kube-Proxy in the garden/seed cluster should be configured.

Reply:
Can you elaborate in more details what you mean for `IPVS`?

This sentence tries to explain that in `iptables` mode traffic is distributed randomly across all endpoints, hence the traffic is not zone/topology aware. I don't know what is the used algorithm in `ipvs` mode but I doubt that it is zone/topology aware. Still, it make sense to check it to be 100% sure.

Reply:
Okay, thanks for making it clear. Seems like I misunderstood this section 🙂